modules/web-console/web-console-server/src/test/java/org/apache/ignite/console/listener/NotificationEventListenerTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.console.listener;
+
+import org.apache.ignite.console.TestConfiguration;
+import org.apache.ignite.console.dto.Account;
+import org.apache.ignite.console.event.Event;
+import org.apache.ignite.console.event.EventPublisher;
+import org.apache.ignite.console.notification.NotificationDescriptor;
+import org.apache.ignite.console.services.NotificationService;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.apache.ignite.console.event.Type.ACCOUNT_CREATE;
+import static org.apache.ignite.console.event.Type.ACCOUNT_CREATE_BY_ADMIN;
+import static org.apache.ignite.console.event.Type.ACCOUNT_DELETE;
+import static org.apache.ignite.console.event.Type.PASSWORD_CHANGED;
+import static org.apache.ignite.console.event.Type.PASSWORD_RESET;
+import static org.apache.ignite.console.event.Type.RESET_ACTIVATION_TOKEN;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Notification event listener test.
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = {TestConfiguration.class})
+public class NotificationEventListenerTest {
+    /** Publisher. */
+    @Autowired
+    private EventPublisher publisher;
+
+    /** Notification Server. */
+    @MockBean
+    private NotificationService notificationSrv;
+
+    /**
+     * Test welcome letter sending
+     */
+    @Test
+    public void testOnUserCreateEvent() throws InterruptedException {
+        Account acc = new Account();
+
+        publisher.publish(new Event<>(ACCOUNT_CREATE, acc));
+        Thread.sleep(1);
```

Comment:
Why we need sleep here? I think this can be removed

Reply:
Change to sync executor for test.